roles/icinga_agent/tasks/main.yml

```diff
@@ -86,29 +88,38 @@
         group: "{{ icinga2_group[ansible_os_family] }}"
 
     - name: generate ticket and save it as a variable
-      ansible.builtin.shell: /usr/sbin/icinga2 pki ticket --cn {{ ansible_hostname }} --salt {{ icinga_agent_salt }}
+      ansible.builtin.command: >
+        /usr/sbin/icinga2 pki ticket --cn {{ ansible_hostname }} --salt {{ icinga_agent_salt }}
       environment:
         LD_LIBRARY_PATH: "/usr/lib64"
       register: ticket
+      changed_when: ticket.rc != 0
+      failed_when: ticket.rc != 0
```

Comment:
the `changed_when` and `failed_when` conditions are the same, this looks wrong to me. Can you explain?

Reply:
This is to pass the ansible-lint check no-changed-when: `Commands should not change things if nothing needs doing`. 

If the Returncode of the command is 0 than the task is `ok` and not `changed`. Is the Returncode of the command is 0 then the task is not failed. 

```
TASK [t_systems_mms.ansible_collection_icinga.icinga_agent : generate ticket and save it as a variable] *********************************************************************************************************
task path: /home/foo/my-ansible/collections/ansible_collections/t_systems_mms/ansible_collection_icinga/roles/icinga_agent/tasks/main.yml:90
ok: [my-example-host] => changed=false
  cmd:
  - /usr/sbin/icinga2
  - pki
  - ticket
  - --cn
  - my-example-host
  - --salt
  - xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
  delta: '0:00:00.267765'
  end: '2022-08-17 09:44:58.978878'
  failed_when_result: false
  rc: 0
  start: '2022-08-17 09:44:58.711113'
  stderr: ''
  stderr_lines: <omitted>
  stdout: <my ticket> 
  stdout_lines: <omitted>
```

So both values are correct. 

See also: https://github.com/ansible/ansible-lint/issues/1681

Reply:
I just checked with the Icinga2 repository: https://github.com/Icinga/icinga2/blob/ee705bb110e802f8cafd21bab2d8697b0a538b0a/lib/cli/pkiticketcommand.cpp#L37-L55

It seems they only generate a different exit code, if there was an error. My concern is, that using `changed_when`suggests, that this task could display a changed status in execution. I think in this case I would prefer to use `changed_when: false` to signify, that this task will never display a changed status.

Reply:
Or disable the specific ansible-lint rule for this one task.

Reply:
I changed it to false: https://github.com/T-Systems-MMS/ansible-collection-icinga/pull/31/commits/2b7a467eec09a269cb2b41d5cfba2897bfb46e6a